mavros/src/plugins/waypoint.cpp

```diff
@@ -28,67 +32,168 @@
 
 namespace mavros {
 namespace std_plugins {
-using utils::enum_value;
 using mavlink::common::MAV_CMD;
 using mavlink::common::MAV_FRAME;
 using MRES = mavlink::common::MAV_MISSION_RESULT;
+using MAV_CAP = mavlink::common::MAV_PROTOCOL_CAPABILITY;
+using utils::enum_value;
+using mavlink::common::MAV_FRAME;
+using WP_ITEM = mavlink::common::msg::MISSION_ITEM;
+using WP_ITEM_INT = mavlink::common::msg::MISSION_ITEM_INT;
+
+double waypoint_encode_factor( const uint8_t &frame ){
+	// [[[cog:
+	//from pymavlink.dialects.v20 import common
+	//e=common.enums['MAV_FRAME']
+	//all_names = [ee.name[len('MAV_FRAME_'):] for ee in e.values()]
+	//all_names.pop() # remove ENUM_END
+	//global_names = [v for v in all_names if v.startswith('GLOBAL')]
+	//local_names = [v for v in all_names if v.startswith(('LOCAL', 'BODY', 'MOCAP', 'VISION', 'ESTIM'))]
+	//other_names = ['MISSION']
+	//cog.outl("switch(frame){")
+	//for names, factor in [(global_names, 10000000), (local_names, 10000), (other_names, 1)]:
+	//	for name in names:
+	//		cog.outl(f"case enum_value(MAV_FRAME::{name}):")
+	//	cog.outl(f"\treturn {factor};")
+	//cog.outl("default:\n\treturn 1;")
+	// ]]]
+	switch(frame){
+	case enum_value(MAV_FRAME::GLOBAL):
+	case enum_value(MAV_FRAME::GLOBAL_RELATIVE_ALT):
+	case enum_value(MAV_FRAME::GLOBAL_INT):
+	case enum_value(MAV_FRAME::GLOBAL_RELATIVE_ALT_INT):
+	case enum_value(MAV_FRAME::GLOBAL_TERRAIN_ALT):
+	case enum_value(MAV_FRAME::GLOBAL_TERRAIN_ALT_INT):
+		return 10000000;
+	case enum_value(MAV_FRAME::LOCAL_NED):
+	case enum_value(MAV_FRAME::LOCAL_ENU):
+	case enum_value(MAV_FRAME::LOCAL_OFFSET_NED):
+	case enum_value(MAV_FRAME::BODY_NED):
+	case enum_value(MAV_FRAME::BODY_OFFSET_NED):
+	case enum_value(MAV_FRAME::BODY_FRD):
+	case enum_value(MAV_FRAME::BODY_FLU):
```

Comment:
Problem with that two frames arises because upstream mavlink changed recently.
I think that you just need to reinstall/update pymavlink to latest one, then run cog.

Reply:
Ok, so I updated both my upstream mavlink and pymavlink and it didn't change, then I installed pymavlink from the submodule in mavlink and it did work so I'm thinking they haven't fixed it in the pip repository yet? But the code should be correct now and hopefully they will update it by the time it needs to be generated again.

Reply:
Yes. As i remember pip releases does @tridge with different schedule.
I prefer to use names and not hardcode values, so yes, that code would require update later.
But because we keep generator code, that should be easy.